common/lib/xmodule/xmodule/graders.py

```diff
@@ -81,25 +82,58 @@ def float_sum(iterable):
 
 def aggregate_scores(scores):
     """
-    scores: A list of ScoreBase objects
+    scores: A list of ProblemScore objects
     returns: A tuple (all_total, graded_total).
-        all_total: A ScoreBase representing the total score summed over all input scores
-        graded_total: A ScoreBase representing the score summed over all graded input scores
+        all_total: An AggregatedScore representing the total score summed over all input scores
+        graded_total: An AggregatedScore representing the score summed over all graded input scores
     """
-    total_correct_graded = float_sum(score.earned for score in scores if score.graded)
-    total_possible_graded = float_sum(score.possible for score in scores if score.graded)
-    any_attempted_graded = any(score.attempted for score in scores if score.graded)
+    def _iter_graded(scores):
```

Comment:
Yeah.  I'm generally in favor of keeping them outside for the stated performance reason.  If it's inside a method, a helper method is a good alternative (still namespaced in the class, but created at runtime).  

@efischer19 I'd say the best time to create an inner function is "rarely," or to be more specific, "when you can specifically justify doing it that way."